controllers/openstackmachine_controller.go

```diff
@@ -527,6 +527,9 @@ func openStackMachineSpecToOpenStackServerSpec(openStackMachineSpec *infrav1.Ope
 				},
 			}
 		}
+		if len(openStackMachineSpec.SecurityGroups) > 0 {
```

Comment:
nit: This if statement is redundant. The append below works correctly when SecurityGroups is empty.

Reply:
This one always gets me :joy:

Reply:
oh I could have fixed it. But it's already merged and backported.
Thanks